sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeActionsOperation.cs

```diff
@@ -281,33 +298,26 @@ async ValueTask<OperationState<AsyncPageable<AnalyzeActionsResult>>> IOperation<
 
             Response rawResponse = response.GetRawResponse();
 
-            if (response.Value.Status == TextAnalyticsOperationStatus.Failed)
-            {
-                if (CheckIfGenericError(response.Value))
-                {
-                    RequestFailedException requestFailedException;
-
-                    if (async)
-                    {
-                        requestFailedException = await ClientCommon.CreateExceptionForFailedOperationAsync(true, _diagnostics, rawResponse, response.Value.Errors).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        requestFailedException = ClientCommon.CreateExceptionForFailedOperationAsync(false, _diagnostics, rawResponse, response.Value.Errors).EnsureCompleted();
-                    }
-
-                    return OperationState<AsyncPageable<AnalyzeActionsResult>>.Failure(rawResponse, requestFailedException);
-                }
-            }
-
-            if (response.Value.Status == TextAnalyticsOperationStatus.Succeeded ||
-                response.Value.Status == TextAnalyticsOperationStatus.Failed)
+            if (response.Value.Status == TextAnalyticsOperationStatus.Succeeded)
             {
                 string nextLink = response.Value.NextLink;
                 _firstPage = Page.FromValues(new List<AnalyzeActionsResult>() { response.Value.Result }, nextLink, rawResponse);
```

Comment:
>_firstPage [](http://example.com/codeflow?start=16&length=10)

Why do we set this as part of this function call? We seem to be accessing _firstPage as part of GetVaules however it is getting updated here which is a bit confusing.

Reply:
`GetValues` uses `_firstPage` that gets set here because `UpdateStatus/Async` is designed to set everything - even the final value if the LRO is successful.